artemis/modules/base/module_configuration.py

```diff
@@ -0,0 +1,56 @@
+from typing import Any, ClassVar, Dict, Type, TypeVar, cast
+
+T = TypeVar("T", bound="ModuleConfiguration")
+
+
+class ModuleConfiguration:
+    """
+    Base class for all module-specific configurations in Artemis.
+
+    This class provides a standardized interface for module configurations with
+    serialization, deserialization, and validation capabilities.
+
+    Attributes:
+        enabled (bool): Whether the module is enabled. Defaults to True.
+    """
+
+    def __init__(self, enabled: bool = True) -> None:
+        """
+        Initialize a new ModuleConfiguration instance.
+
+        Args:
+            enabled (bool, optional): Whether the module is enabled. Defaults to True.
+        """
+        self.enabled = enabled
+
+    def serialize(self) -> Dict[str, Any]:
+        """
+        Serialize the configuration to a dictionary.
+
+        Returns:
+            Dict[str, Any]: Dictionary representation of the configuration.
+        """
+        return {"enabled": self.enabled}
```

Comment:
the decision whether modules are enables is made earlier, so "enabled" shouldn't be there, IMO you can return an empty dictionary